Code/Framework/AzNetworking/AzNetworking/ConnectionLayer/ConnectionEnums.h

```diff
@@ -34,6 +34,22 @@ namespace AzNetworking
         , Connecting
     );
 
+    inline const char* GetEnumString(ConnectionState connectionState)
+    {
+        switch(connectionState)
+        {
+        case ConnectionState::Disconnected:
+            return "Disconnected";
+        case ConnectionState::Disconnecting:
+            return "Disconnecting";
+        case ConnectionState::Connected:
+            return "Connected";
+        case ConnectionState::Connecting:
+            return "Connecting";
+        }
+        return "INVALID";
+    }
```

Comment:
ConnectionState is an AZ_ENUM_CLASS which defines a constexpr ToString. You can just use that.

Reply:
fixed!